omod/src/main/java/org/openmrs/module/patientqueueing/page/controller/ProviderDashboardPageController.java

```diff
@@ -0,0 +1,77 @@
+import org.openmrs.module.appframework.context.AppContextModel;
+import org.openmrs.module.appframework.domain.AppDescriptor;
+import org.openmrs.module.appframework.domain.Extension;
+import org.openmrs.module.appframework.service.AppFrameworkService;
+import org.openmrs.module.appui.UiSessionContext;
+import org.openmrs.module.coreapps.CoreAppsProperties;
+import org.openmrs.module.emrapi.adt.AdtService;
+import org.openmrs.module.emrapi.event.ApplicationEventService;
+import org.openmrs.ui.framework.BasicUiUtils;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.page.PageModel;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.openmrs.ui.framework.UiUtils;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ProviderDashboardPageController {
+	
+	public Object controller(PageModel model, @RequestParam(required = false, value = "apps") AppDescriptor app,
+	        @RequestParam(required = false, value = "dashboard") String dashboard,
+	        @SpringBean("adtService") AdtService adtService,
+	        @SpringBean("appFrameworkService") AppFrameworkService appFrameworkService,
+	        @SpringBean("applicationEventService") ApplicationEventService applicationEventService,
+	        @SpringBean("coreAppsProperties") CoreAppsProperties coreAppsProperties, UiSessionContext sessionContext) {
+		
+		if (StringUtils.isEmpty(dashboard)) {
+			dashboard = "providerDashboard";
+		}
+		model.addAttribute("apps", app);
+		UiUtils uiUtils = new BasicUiUtils();
+		
+		AppContextModel contextModel = sessionContext.generateAppContextModel();
+		model.addAttribute("appContextModel", contextModel);
+		
+		List<Extension> includeFragments = appFrameworkService.getExtensionsForCurrentUser(dashboard + ".includeFragments",
+		    contextModel);
+		Collections.sort(includeFragments);
+		model.addAttribute("includeFragments", includeFragments);
+		
+		List<Extension> rightColumnFragments = appFrameworkService.getExtensionsForCurrentUser(dashboard
+		        + ".rightColumnFragments", contextModel);
+		Collections.sort(rightColumnFragments);
+		model.addAttribute("rightColumnFragments", rightColumnFragments);
+		
+		List<Extension> leftColumnFragments = appFrameworkService.getExtensionsForCurrentUser(dashboard
+		        + ".leftColumnFragments", contextModel);
+		Collections.sort(leftColumnFragments);
+		model.addAttribute("leftColumnFragments", leftColumnFragments);
+		
+		List<Extension> otherActions = appFrameworkService.getExtensionsForCurrentUser(
+		    (dashboard == "providerDashboard" ? "provider" : dashboard) + ".otherActions", contextModel);
+		Collections.sort(otherActions);
+		model.addAttribute("otherActions", otherActions);
+		
+		model.addAttribute("baseDashboardUrl", coreAppsProperties.getDashboardUrl()); // used for breadcrumbs to link back to the base dashboard in the case when this is used to render a context-specific dashboard
```

Comment:
Could you add this comment at the top instead of on the right hand side?

Reply:
Resolved in #6